src/TypeSpec/__Private/UntypedArraySpec.hack

```diff
@@ -24,7 +24,7 @@ final class UntypedArraySpec extends TypeSpec<array> {
         TypeCoercionException::withValue($this->getTrace(), 'array', $value);
     }
 
-    $out = [];
+    $out = darray[];
```

Comment:
Given that this is `coerce` and not `assert`, it's probably fine to change the type of the value (arbitrary array/varray/darray to always darray).

I tried to look at the other TypeSpec to see if there are any other examples of coercion possibly changing the value's type, but couldn't find any. The only ones where it would be applicable (they actually have multiple subtypes) are TraversableSpec and KeyedTraversableSpec, but those are declared as `NoCoercionSpecTrait`, not sure why.

Reply:
Shapes - even in assertion, dict -> darray so that it can be used with json_decode